File: frontend/src/objects/auth/PersistedAuthAccount.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  AuthUserId,
  EntityId,
  IsoDateTime,
  PasswordHash,
  PersonName,
  UserRole,
  Username,
} from '@/objects/core/SharedObjects'

export type PersistedAuthAccount = {
  id: AuthUserId
  username: Username
  passwordHash: PasswordHash
  role: UserRole
  displayName: PersonName
  linkedProfileId?: EntityId
  createdAt: IsoDateTime
}

File: backend/src/main/scala/services/auth/objects/PersistedAuthAccount.scala
package services.auth.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class PersistedAuthAccount(
    id: AuthUserId,
    username: Username,
    passwordHash: PasswordHash,
    role: UserRole,
    displayName: PersonName,
    linkedProfileId: Option[EntityId],
    createdAt: IsoDateTime,
)
object PersistedAuthAccount:
  given Encoder[PersistedAuthAccount] = deriveEncoder
  given Decoder[PersistedAuthAccount] = deriveDecoder
